### ISSUE TITLE
SetCorsorPos - does not work correctly!

### ISSUE THREAD

**Issue by ElectroidDes** [ocornut/imgui#7900]
### Version/Branch of Dear ImGui:

1.90.8

### Back-ends:

Raylib/

### Compiler, OS:

Windows10

### Details:

SetCursorPos incorrectly positions objects.

Below is the code and screenshots. For the first button, I position the Upper Left corner to the GetWindowContentRegionMin coordinate. As you can see in the screenshot, everything happens correctly.

HOWEVER, when I position the Upper Left corner of the second button to the GetWindowContentRegionMax coordinate, SetCursorPos misses by one pixel!

 ```
 bool bool_;

  ImGui::Begin("Main Window", &bool_, ImGuiWindowFlags_NoScrollbar);

  //----------------------------------------------------------------------------------------------------------
  ImVec2 WindowContentRegionMin_comp = { ImGui::GetWindowContentRegionMin().x + ImGui::GetScrollX() + ImGui::GetWindowPos().x, ImGui::GetWindowContentRegionMin().y + ImGui::GetScrollY() + ImGui::GetWindowPos().y };
  ImVec2 WindowContentRegionMax_comp = { ImGui::GetWindowContentRegionMax().x + ImGui::GetScrollX() + ImGui::GetWindowPos().x, ImGui::GetWindowContentRegionMax().y + ImGui::GetScrollY() + ImGui::GetWindowPos().y };

  ImDrawList* draw_list = ImGui::GetWindowDrawList();

  draw_list->AddRect(WindowContentRegionMin_comp, WindowContentRegionMax_comp, IM_COL32(255, 0, 255, 255));
  //----------------------------------------------------------------------------------------------------------

  ImGui::SetCursorPos(ImGui::GetWindowContentRegionMin());
  ImGui::Button("Button_1", { 50,50 });


  ImGui::SetCursorPos(ImGui::GetWindowContentRegionMax());
  ImGui::Button("Button_2", { 50,50 });



  ImGui::End();
```

![image](https://github.com/user-attachments/assets/42cbabfd-d6e9-47c5-ad8d-333f3d1afe88)

**Comment by ocornut** [comment]
Please label your issues correctly. Clearly this has nothing to do with SetCursorPos().

What is wrong about this? 
It seems that AddRect being min-inclusive and max-exclusive here is confusing the situation.

You have been posting dozens of bizarre, overly precise issues. What are you trying to do and to solve?

**Comment by ocornut** [comment]
(Also please consider updating to 1.91.0 and not using those local-coordinate functions which have been obsoleted because don’t bring value but mostly confusion)

**Comment by ElectroidDes** [comment]
> Please label your issues correctly. Clearly this has nothing to do with SetCursorPos().
> 
> What is wrong about this? It seems that AddRect being min-inclusive and max-exclusive here is confusing the situation.
> 
> You have been posting dozens of bizarre, overly precise issues. What are you trying to do and to solve?

I am pointing out an obvious problem.
If you think that incorrect positioning of an object using SetCursorPos is normal, then let this incorrect behavior remain in Imgui, I don't mind :)

**Comment by ElectroidDes** [comment]
> GetWindowContentRegionMin

For me, GetWindow ContentRegionMin-GetWindowContentRegionMax are useful. With their help, I define the content rectangle in windows.

**Comment by ocornut** [comment]
> If you think that incorrect positioning of an object using SetCursorPos is normal, then let this incorrect behavior remain in Imgui, I don't mind :)

The problem is not in SetCursorPos() the problem might be in the value passed to it. This is like passing a wrong value to printf() and stating that printf() doesn’t work.

You have not answered my question. What are you trying to solve and do with all those dozen questions relative to micro pixel positioning of items?

**Comment by ocornut** [comment]
> For me, GetWindow ContentRegionMin-GetWindowContentRegionMax are useful. With their help, I define the content rectangle in windows.

As stated in 1.91 release note you can use GetCursorScreenPos() and GetContentRegionAvail() after Begin() and obtain the same values in absolute coordinates.

What’s the obvious problem?

I don’t see even a problem here, it looks correct to me. It is the AddRect call that draw rectangles this way.

If you don’t mind what you suggest is an “obviously incorrect behavior” not being fixed then please consider not posting as many issues, this may be hogging development resources. We want to solve real blocking problems that people have. If you have a problem we’ll help.

**Comment by ElectroidDes** [comment]
> You have not answered my question. What are you trying to solve and do with all those dozen questions relative to micro pixel positioning of items?

This is a very strange question, frankly speaking. I am trying to make my widgets position correctly, and not arbitrarily. What kind of library is this that positions objects differently each time? It is difficult to do anything with such a mechanism - it is better to go and drown yourself.

Changed the frame width - all widgets "moved".
In one case, SetCursorPos positions correctly, in another it does not.

I gave a specific example: I position the Upper Left corner of the button to the GetWindowContentRegionMax coordinate and the screenshot shows that the positioning is incorrect. What is your question then?
